coa-helm/templates/tes.yaml

```diff
@@ -1,75 +1,159 @@
-#            - name: DisableBatchScheduling
-#              value: {{ .Values.config.disableBatchScheduling | quote }}
-#            - name: DockerInDockerImageName
-#              value: {{ .Values.config.dockerInDockerImageName }}
-#            - name: UsePreemptibleVmsOnly
-#              value: {{ .Values.config.usePreemptibleVmsOnly | quote}}
-#            - name: MarthaUrl
-#              value: {{ .Values.config.drsUrl }}
-#            - name: CromwellDrsLocalizerImageName
-#              value: {{ .Values.cromwell.image | replace "cromwell" "cromwell-drs-localizer" }}
-#          image: {{ .Values.tes.image }}
-#          name: tes
-#          ports:
-#            - containerPort: {{ .Values.tes.port }}
-#          resources: {}
-#      restartPolicy: Always
-#status: {}
-#
-#---
-#
-#apiVersion: v1
-#kind: Service
-#metadata:
-#  labels:
-#    io.kompose.service: tes
-#  name: tes
-#spec:
-#  ports:
-#    - name: {{ .Values.tes.port | quote }}
-#      port: {{ .Values.tes.port }}
-#      targetPort: {{ .Values.tes.port }}
-#  selector:
-#    io.kompose.service: tes
-#status:
-#  loadBalancer: {}
-#
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    {{- include "app.labels" . | nindent 4 }}
+  name: {{ include "app.fullname" . }}-tes-depl
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      io.kompose.service: tes
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        io.kompose.service: tes
+        aadpodidbinding: {{ .Values.identity.name }}
+    spec:
+      containers:
+        - env:
+            # Name needs to be unique per LZ with max length 9. coa-ABCDEF-coa-rls -> ABCDEF
+            # Caution!! Consider hardening this to ensure uniqueness.
+            - name: Name
+              value: {{ .Values.fullnameOverride | substr 4 10 }}
+            - name: ApplicationInsights_Connection_String
+              value: {{ .Values.config.applicationInsightsAccountKey }}
+            - name: AzureOfferDurableId
+              value: {{ .Values.tes.azureOfferDurableId }}
+            - name: BatchAccount__AccountName
+              value: {{ .Values.config.batchAccountName }}
+            - name: BatchAccount__BaseUrl
+              value: "https://{{ .Values.config.batchAccountName }}.{{ .Values.config.region }}.batch.azure.com"
+            - name: BatchAccount__AppKey
+              value: {{ .Values.config.batchAccountKey }}
+            - name: BatchAccount__Region
+              value: {{ .Values.config.region }}
+            - name: BatchAccount__SubscriptionId
+              value: {{ .Values.config.subscriptionId }}
+            - name: BatchAccount__ResourceGroup
+              value: {{ .Values.config.resourceGroup }}
+            # Not currently working
+            # - name: BatchNodesSubnetId
+            #   value: {{ .Values.config.batchNodesSubnetId }}
+            - name: RetryPolicyOptions__MaxRetryCount
```

Comment:
This has been renamed to start with `RetryPolicy__` in the latest Develop branch: https://github.com/microsoft/ga4gh-tes/blob/304b69af5ca302f05356561c9f5e3af4d86fd774/src/TesApi.Web/Management/Configuration/RetryPolicyOptions.cs#L14